.github/workflows/deny_dirty_cargo_locks.yml

```diff
@@ -0,0 +1,32 @@
+name: Check no Cargo.lock files are dirty
+
+on:
+  [push, pull_request]
+
+jobs:
+  no_dirty_cargo_locks_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
By default it checks out the head of the PR I don't think this needs to be set manually.

Reply:
It seems to checkout the branch on the fork by default, but to compare against main below we need it to checkout the head in _this_ repository.

Reply:
I think this behavior is okay, especially since we don't actually allow merge commits.

Reply:
We have temporary merge commits due to people clicking "update branch"